src/__tests__/components/AutomationDetails.test.jsx

```diff
@@ -64,27 +62,121 @@ const props = {
   },
 };
 
-const getComponent = () => shallow(<AutomationDetails {...props} />);
+const getComponent = () => mount(<AutomationDetails {...props} />);
 
 describe('Automation details', () => {
   it('should render slack details', () => {
-    expect(getComponent()).toMatchSnapshot();
+    const component = getComponent();
+    expect(component.instance().state.modalType).toEqual('slack');
   });
 
   it('should render email details', () => {
     const component = getComponent();
-    component.setProps({
-      modalType: 'email',
-    });
-    expect(getComponent()).toMatchSnapshot();
-    expect(component.find('.table-body-details')).toBeTruthy();
+    component.find('#email').simulate('click');
+    expect(component.instance().state.modalType).toEqual('email');
   });
 
   it('should render freckle details', () => {
     const component = getComponent();
-    component.setProps({
-      modalType: 'freckle',
-    });
-    expect(getComponent()).toMatchSnapshot();
+    component.find('#freckle').simulate('click');
+    expect(component.instance().state.modalType).toEqual('freckle');
+  });
+
+  it('should render freckle details with no activities', () => {
+    const prop = {
```

Comment:
The `props` variable at line 5, this `prop` variable and the one at line 141 have duplicate content, please employ the `component.instance.setProps({})` method to set component instance props. This will avoid this repetition of props content and make the code DRY.